Xwt/Xwt/TreePosition.cs

```diff
@@ -26,13 +26,22 @@
 
 
 using System;
-using Xwt.Drawing;
 using System.Collections.ObjectModel;
 using System.Collections.Generic;
+using Xwt.Accessibility;
 
 namespace Xwt
 {
-	public interface TreePosition
+	public class TreePosition
 	{
+		Accessible accessible;
+		public Accessible Accessible {
```

Comment:
This is not right. TreePosition is just a reference to a position, it can't contain data itself.

Reply:
@slluis,
Hmm. I think alternatively I can add a11y API to `ITreeStoreBackend`, which would look like this:
```
ITreeStoreBackend {
    Accessible GetChildAccessibility (TreePosition child);
} 
```
Does it look good to you?

Reply:
@slluis I realized it might be not a good idea to instantiate Accessibility on the backend side, moreover its constructors are internal.

The problem is I need to send AutomationPeer somehow to the WPF backend, using `TreePosition`, `TreeNavigator` or `TreeStore` APIs. If I don't use TreePosition, then I should use `TreeStore` or`TreeNavigator` to send it to ITreeStoreBackend, where I would attach the accessible object to the proper Child (TreeStoreNode in case of Xwt.WPF). The API would like like this `TreeNavigator.AddAccessibleChild (object nativeChild)` (and maybe `RemoveAccessibleChild`), which would be used to deliver `AutomationPeer` to `TreeStoreNode` and then to `ExTreeViewItem`.

Or also I can use `TreePosition` interface for the same purpose -- `TreePosition.AddAccessibleChild (object nativeChild)`. This way the accessible object would be delivered directly to `TreeStoreNode`, without redirecting this work to `TreeStoreBackend`.
What do you think?

Reply:
I'm not convinced that this is the right approach in general. `TreePosition`, `TreeNavigator` or `TreeStore` are handling the data and not it's representation, hence they can not be accessible in any way, `TreeView` with it's `CellView`s is the visualization and can/should have accessibility features.

The right approach might be to add `Accessible` and `AccessibleField` properties to the `CellView`s. It could be either a `DataField<Accessible>` or we could map several `Accessible*` fields (AccessibleLabelField, etc.). this would allow us to control accessibility on cell level rather than on rows (your WPF specific approach is row based, since `ExTreeViewItem` is a single row representation). This is also how Cocoa accessibility works (setting the props on single NSViews in view based tree mode).

Reply:
I agree with @sevoku. I think a good approach is to have an AccessibleFields class in CellView, which would have one field for each accessible property. Something like this:

``` c#
public class CellView {
    ...
    public AccessibleFields AccessibleFields { get; } = new AccessibleFields ();
    ...
}

public class AccessibleFields
{
    public DataField<string> Label = new DataField<string> ();
    public DataField<string> Title = new DataField<string> ();
    ....
}
```

It would be better to crate all this on demand though.

Reply:
@sevoku Hm, but TreeView doesn't expose any `CellView`s. And even if we can get `CellView` somehow, settings a11y to specific CellView inside the row wouldn't work for the whole row. We are selecting the whole row and see its a11y properties and not its internal UI components. So user would need to go inside the row through its components to get a picture what is presented in that row.  I might get something wrong here, correct me if so.
Getting `CellView` might be related to the `FrameworkElementFactory`, which is used to create the column cell template: https://github.com/mono/xwt/blob/master/Xwt.WPF/Xwt.WPFBackend/TreeViewBackend.cs#L261. Not sure how exactly, but maybe by listening events from this factory. And then we should expose these `CellView`s to the Xwt client using something like Xwt.CellView as a fronted representation. Is it what you meant?
If so, don't we need to set `Accessibility` to ExTreeViewItem as well? That would mean that we need to expose TreeViewItem's frontend like Xwt.TreeViewItem?

Reply:
The thing is that a row can contain several completely independent views. Not only column-wise, also every column can contain several CellCiews inside. Pls check the samlples especially here: https://github.com/mono/xwt/blob/master/TestApps/Samples/Samples/TreeViews.cs#L48-L102.
So if a user selects a row and we present "some" accessible text for it, it can't reflect all the information inside the particular row, since it can contain various different components, represented ba different `CellView` types (besides a label, it can be a checkbox or a combo box or whatever).
In Cocoa and also WPF a row contains different views, arranged either by columns or specialized views inside a column (a column can contain several different view types). Each view (text, icon, check, combo,...) has different purposes and information. Hence it is not sufficient to present just a row-based information, which does not allow the user to interact with different components and also get column specific information.
@DmitriyKirakosyan you are talking about WPF specific "hacks", to make single rows accessible, which is not sufficient to represent all data in a view context and is not compatible with other toolkits (only Xwt.WPF is using a special row-view) . And what you propose is data-driven and not view related.

Reply:
@sevoku Thanks for the explanation! It does make sense to me now. Just to make sure I understand you correctly:
I add `AccessibleFields` to CellView, then on the client side I add a11y fields to a `CellView` object and to `TreeStore` here https://github.com/mono/xwt/blob/master/Xwt/Xwt/TreeStore.cs#L59, and then set values here https://github.com/mono/xwt/blob/master/Xwt/Xwt/TreeNavigator.cs#L119
Right?

Reply:
Yeah, this is exactly what I mean :) So it works like all the other CellView data/properties.